### PR TITLE
Isolate CEN logic, components wiring

### PR DESCRIPTION
**Pull request by i-schuetz**

### CODE REVIEW

**Comment by i-schuetz** [comment]
@rodneywitcher I put all the CEN logic in a file (unsurprisingly named `CenLogic`) and refactored a bit to make the flow clearer. Could you please go quickly through this file and confirm everything is correct?

File: CoEpi/domain/logic/CenLogic.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import Foundation
+import CryptoSwift
+
+class CenLogic {
```
Comment (johnsonh) [inline CoEpi/domain/logic/CenLogic.swift:4]: CenGenerator?

File: CoEpi/domain/logic/CenLogic.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        //convert timestamp to [UInt8]
+        var tsAsUInt8Array: [UInt8] = []
+        [roundedTimestamp(ts: currentTs)].withUnsafeBytes {
+            tsAsUInt8Array.append(contentsOf: $0)
+        }
+
+        //encrypt tsAsUnit8Array using decodedCENKey... using AES
+        let encData = try! AES(key: decodedCENKeyAsUInt8Array, blockMode: ECB(), padding: .pkcs5).encrypt(tsAsUInt8Array)
```
Comment (johnsonh) [inline CoEpi/domain/logic/CenLogic.swift:44]: Could reduce all of this to a computed var  on CENKey, or have a toData on CEN

Comment (duskoo) [reply CoEpi/domain/logic/CenLogic.swift:44]: CenLogic could be part of CenKey class, but this also works.

File: CoEpi/domain/logic/CenLogic.swift (hunk 1 above)
Comment (i-schuetz) [reply CoEpi/domain/logic/CenLogic.swift:4]: It does more...

File: CoEpi/domain/logic/CenLogic.swift (hunk 2 above)
Comment (i-schuetz) [reply CoEpi/domain/logic/CenLogic.swift:44]: That could work, though what if we wanted to try out different generation algorithms? We wouldn't be able to swap those easily.

I'd perhaps consider splitting this class in `CenKeyGenerator` and `CenGenerator`?

**Pull request merged**